test/offline/framework/orchestration.cpp

```diff
@@ -0,0 +1,289 @@
+#include "orchestration.h"
+
+#include "execution.h"
+#include "file_management.h"
+
+#include "common/log/log.h"
+#include "common/util/FileUtil.h"
+#include "common/util/StringUtil.h"
+#include "common/util/diff.h"
+
+#include "decompiler/ObjectFile/ObjectFileDB.h"
+#include "test/offline/config/config.h"
+
+#include "third-party/fmt/color.h"
+#include "third-party/fmt/core.h"
+#include "third-party/fmt/ranges.h"
+
+// TODO - this should probably go somewhere common when it's needed eventually
+std::unordered_map<std::string, std::string> game_name_to_config = {
+    {"jak1", "jak1_ntsc_black_label.jsonc"},
+    {"jak2", "jak2_ntsc_v1.jsonc"}};
+
+OfflineTestThreadManager g_offline_test_thread_manager;
+
+OfflineTestDecompiler setup_decompiler(const OfflineTestWorkGroup& work,
+                                       const fs::path& iso_data_path,
+                                       const OfflineTestConfig& offline_config) {
+  // TODO - pull out extractor logic to determine release into common and use here
+  OfflineTestDecompiler dc;
+  dc.config = std::make_unique<decompiler::Config>(
+      decompiler::read_config_file((file_util::get_jak_project_dir() / "decompiler" / "config" /
+                                    game_name_to_config[offline_config.game_name])
+                                       .string()));
+
+  // TODO - do I need to limit the `inputs.jsonc` as well, or is the decompiler smart enough
+  // to lazily load the DGOs as needed based on the allowed objects?
+
+  // modify the config
+  std::unordered_set<std::string> object_files;
+  for (const auto& coll : work.work_collections) {
+    for (auto& file : coll.source_files) {
+      object_files.insert(file.name_in_dgo);  // todo, make this work with unique_name
+    }
+    for (auto& file : coll.art_files) {
+      object_files.insert(file.unique_name);
+    }
+  }
+
+  dc.config->allowed_objects = object_files;
+  // don't try to do this because we can't write the file
+  dc.config->generate_symbol_definition_map = false;
+
+  std::vector<fs::path> dgo_paths;
+  for (auto& x : offline_config.dgos) {
+    dgo_paths.push_back(iso_data_path / x);
```

Comment:
I think you could trim down this list here to only included dgos that are in the work group.